### PR TITLE
Add images to example page

### LINKED ISSUE #17692: Suggestion: Add image thumbnails to examples menu
##### Thumbnails would make browsing the examples more useful

I love spending time browsing the examples and getting inspiration or ideas for my next sketch or experiment but always find it difficult to remember what each one does. 

I made a test that optionally displays an image thumbnail with each entry on the main example page. The images can be shown or hidden using the "Image" button to left next to the search filter. Excuse the hastily thrown together SVG file. 

I wrote a C++ app that embeds Chromium to render each page of examples and capture the output as a JPEG file and I use that for each thumbnail.  It's fully automated so some may not be the best representation of the example but it's straightforward to recapture as necessary.

Is this worth pursuing and if so, I can make a proper pull-request in a fork. 

Thank you.

* [Example](https://3ve.surge.sh/examples/index.html) (I made a demo to illustrate what I mean)

##### Three.js version

- [x] Dev
- [x] r109
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by marcofugaro**
Followup of discussion in #18839

Originally proposed by @looeee and @munrocket, this PR explores a way of adding images directly into the existing sidebar.

It includes a card layout for each link, which I found to be the perfect solution for its extensibility and simple interactivity. It also fits with the current threejs.org design.

Please share your views on this, any feedback is appreciated 🙌

This PR would close #17692

[Screenshot 2020-05-16 at 12 24 29](https://raw.githack.com/marcofugaro/three.js/example-page-images/examples/index.html)

### [Live demo](https://raw.githack.com/marcofugaro/three.js/example-page-images/examples/index.html)

# Possible future improvements
Improvements that might come in future PRs if we agree on it.

## Lazy loading

Downloads only the images currently visible in the viewport. Useful especially for mobile users.

It works with the native [loading="lazy"](https://addyosmani.com/blog/lazy-loading/) attribute, and it follows the Progressive Enhancement methodology, meaning that on unsupported browsers it defaults to the normal loading behaviour ([currently IE and Safari](https://caniuse.com/#feat=loading-lazy-attr)).

## Tags

We could also show to the user the tags introduced in #18779 and #19245. On click of the tag, the search is populated with said tag and all of the examples with said tags are shown.

[Screenshot 2020-05-16 at 12 27 58](https://raw.githack.com/marcofugaro/three.js/example-page-tags/examples/index.html)

## Sticky section titles

The section titles could stick to the top, making it clearer for the user to understand which section he is in.

This would be achieved with the Progressive Enhancement ideology as well, using `position: sticky;` for browsers that support it and automatically defaulting to the normal non-sticky behaviour for browsers that don't support it ([only IE](https://caniuse.com/#feat=css-sticky)).

[Screenshot 2020-05-16 at 12 33 14](https://raw.githack.com/marcofugaro/three.js/example-page-sticky/examples/index.html)

## Grid view
I also explored a grid arrangement for the cards. Having more cards visible makes it quicker for the user to scan through them.

This was also achieved in https://github.com/mrdoob/three.js/pull/18714#issuecomment-591676139, however I think splitting the view horizontally works better than splitting the view vertically, especially for smaller screens (I own a 13'' laptop). Since most examples have the main focus on the center, an horizontal split leaves them more available space (think of it like a square fitted into the view rectangle).

[Screenshot 2020-05-16 at 12 30 08](https://raw.githack.com/marcofugaro/three.js/example-page-grid/examples/index.html)

## Better titles

Currently the titles are formatted like `animation / skinning / blending`, however only the last part is the effective title, `animation` and `skinning` are actually tags. If these are handled like the current tags, we could have a classically formatted title.

Ideally the titles could be more descriptive, containing more than one word.

We could achieve this with another object `var titles =` in the `examples/files.js`.

[Screenshot 2020-05-16 at 12 32 23](https://raw.githack.com/marcofugaro/three.js/example-page-titles/examples/index.html)

### CODE REVIEW

**Comment by munrocket** [comment]
This is great!

Some feedback:
* Lazy loading - I thought this is a default feature.
* Grid view - First mockup in harmony with main page and documentation but users like big grids when they searching examples by pictures. I am suggesting to having both and make a switch like in #18841 but with another [icon](https://fontawesome.com/icons/grip-vertical?style=solid) that works on phones too.

* Tags / Better titles - Current naming system can be rethought from scratch, now it's difficult to make a good design because names too wide, ideally we should have tags on each example and some naming convention ( for example: section > main tag > good name ). Not in first priority but good to see this in future.
* Sticky section titles - have pros an cons

Additional idea to your list:
* [Three.js logo in 3d](https://sketchfab.com/3d-models/threejs-60320862bf904b7ab0e032c27daf7c7c) on first page 😬

I will help you to optimize this page, please say what resolution, format, compression quality and ratio do you like. We can provide a smooth migration.

**Comment by marcofugaro** [comment]
> Tags / Better titles - Current naming system can be rethought from scratch, now it's difficult to make a good design because names too wide, ideally we should have tags on each example and some naming convention ( for example: section > main tag > good name ). Not in first priority but good to see this in future.


Definitely! For the section part of the name, I think we can use the section titles instead, they would replace the current very generic `webgl` section.

> I will help you to optimize this page, please say what resolution, format, compression quality and ratio do you like. We can provide a smooth migration.

Thanks! We can optimize this images if this PR will be merged. I decided to display them in a 8:5 ratio because it looked better than the 4:3 ratio they're in. We could generate the screenshots with that ratio and with smaller dimensions, as long as this doesn't mess up the screenshots matching score.

Thanks you for the remaining feedback!

**Comment by mrdoob** [comment]
Looks good! Many thanks for the explorations 🙏

> ## Lazy loading

Yes, I think we should add this in this PR directly.

> ## Tags

Maybe not needed yet

> ## Sticky section titles

Maybe not needed yet

> ## Grid view

I think we could achieve this by making the panel resizable

> ## Better titles
> Currently the titles are formatted like `animation / skinning / blending`

I was thinking about renaming examples and make the `animation` a subheader. That should clean up the names a bit.

@marcofugaro I feel the screenshot make the scrolling endless now. How about using the screenshot as background of the div and cropping it to half its vertical size?

**Comment by mrdoob** [comment]
@munrocket What do you think about generating screenshots at 400x300 instead of 800x600?

**Comment by munrocket** [comment]
Yea I thought about the screenshots but didn’t come up with a good solution yet :|. Some of the thumbnails is can be black. From the one hand I want to update script that generate screenshot from scratch, from another we will pollute repo. Of course I can try something like #18742 for thumbnails but not sure that it's best case. Ideally we need only one type of screenshots for site and for test. Can we compare it in .jpg? That's the question. Right now we have %20 threshold in each pixel, I think should be ok for diff in tests even for .jpg.

**Comment by mrdoob** [comment]
No no, I didn't mean having two versions.
I meant replacing the current screenshots with smaller versions (400x300 instead of 800x600). So basically changing the script so it generates 400x300 instead.

**Comment by zlf2002171** [comment]
unsubscribe

2020-05-18 

zlf2002171 



发件人："Mr.doob" <notifications@github.com>
发送时间：2020-05-18 13:53
主题：Re: [mrdoob/three.js] Add images to example page (#19375)
收件人："mrdoob/three.js"<three.js@noreply.github.com>
抄送："Subscribed"<subscribed@noreply.github.com>

No no, I didn't mean having two versions.
I meant replacing the current screenshots with smaller versions (400x300 instead of 800x600). So basically changing the script so it generates 400x300 instead.

**Comment by zlf2002171** [comment]
unsubscribe

2020-05-18 

zlf2002171 



发件人："Mr.doob" <notifications@github.com>
发送时间：2020-05-18 13:35
主题：Re: [mrdoob/three.js] Add images to example page (#19375)
收件人："mrdoob/three.js"<three.js@noreply.github.com>
抄送："Subscribed"<subscribed@noreply.github.com>

@munrocket What do you think about generating screenshots at 400x300 instead of 800x600?

**Comment by marcofugaro** [comment]
Alright added the lazy loading.

> @marcofugaro I feel the screenshots make the scrolling endless now. How about cropping the screenshots vertically half size dynamically?

I've updated the CSS aspect ratio of the images, they now have a 16:9 aspect ratio, how does it look?

https://raw.githack.com/marcofugaro/three.js/example-page-images/examples/index.html

>  I think we could achieve this by making the panel resizable

That's a great idea! We could use the `cursor: ew-resize` property on hover of the border to let the user know it's resizable.

**Comment by zlf2002171** [comment]
unsubscribe


2020-05-18 

zlf2002171 



发件人：Marco Fugaro <notifications@github.com>
发送时间：2020-05-18 17:32
主题：Re: [mrdoob/three.js] Add images to example page (#19375)
收件人："mrdoob/three.js"<three.js@noreply.github.com>
抄送："zlf2002171"<zlf2002171@126.com>,"Comment"<comment@noreply.github.com>

Alright added the lazy loading.
@marcofugaro I feel the screenshots make the scrolling endless now. How about cropping the screenshots vertically half size dynamically?
I've updated the CSS aspect ratio of the images, they now have a 16:9 aspect ratio, how does it look?
https://raw.githack.com/marcofugaro/three.js/example-page-images/examples/index.html
I think we could achieve this by making the panel resizable
That's a great idea! We could use the cursor: ew-resize property on hover of the border to let the user know it's resizable.

**Comment by munrocket** [comment]
Ok, I am realized how to fix `webgl_loader_bvh`, `webgl_simple_gi`, `webgl_postprocessing_dof2`, `webgl_physics_volume`, `webgl_loader_fbx_nurbs` screenshots 🚀.

Yeah 16:9 is better, so 400x225? It will give +8.1mb totally. Not much at all.

**Comment by mrdoob** [comment]
@munrocket 

> Yeah 16:9 is better, so 400x225? It will give +8.1mb totally. Not much at all.

I think 400x300 is fine. I wouldn't make the e2e images 16:9, we are just reusing them and cropping them because it suits the UI better.

**Comment by mrdoob** [comment]
Thanks!

**Comment by eXponenta** [comment]
image
There are problems with dark style in Chrome.
Labels text color similar with background color of it, because for dark style palette is inverted.

**Comment by marcofugaro** [comment]
Thanks for the check @eXponenta! Totally missed the `prefers-color-scheme: dark` thing

**Comment by mrdoob** [comment]
@marcofugaro FYI https://github.com/mrdoob/three.js/commit/46d340fa6ffa1c7f0ebfb46a3b5f09b6fefc8128

**Comment by marcofugaro** [comment]
@mrdoob Sure, still looks good! 🙌

Don't forget to bring down also `line-height` when reducing the font size, since it's defined in pixels. A couple pixels down should do the trick!

**Comment by mrdoob** [comment]
@marcofugaro Done https://github.com/mrdoob/three.js/commit/d6f668f6b4e3f5e2401a36118005f50e77a13c9a
The `- 6px` thing seems a bit of a hack though. Maybe it's better to hardcode it to `20px`.

**Comment by greggman** [comment]
this is 1000x less useful than it was with just a list. There are way too many examples to scroll through via thumbnails and the thumbnails themself do not tell you at a glance what the example is. With the text buried between it is now way harder to find what you're looking for.

Please revert!

If you want a solution that might work better leave the list as text on the left and put the thumbnails on the right in a large grid.

**Comment by mrdoob** [comment]
Maybe a checkbox to enable/disable images?

**Comment by marcofugaro** [comment]
@greggman

> There are way too many examples to scroll through via thumbnails 

Maybe #19438 can help?

> the thumbnails themself do not tell you at a glance what the example is

Huh?? Isn't this what thumbnails are for?

> With the text buried between

Between what? Can you elaborate?

**Comment by greggman** [comment]
> @greggman
> 
> > There are way too many examples to scroll through via thumbnails
> 
> Maybe #19438 can help?

That's better but still not as useful as it was

> 
> > the thumbnails themself do not tell you at a glance what the example is
> 
> Huh?? Isn't this what thumbnails are for?

No!

What are these thumbnails about?






I have no idea. The useful info is the text description, not the thumbnail

Thumbnails work for a collection of photos. They do not work for collection of solutions because the import part is not the image, the import part is what feature/tech/solution is being demonstrated. That info is in the text, not the thumbnail

> 
> > With the text buried between
> 
> Between what? Can you elaborate?

As it was before I could at a glance see the titles (ie, descriptions) of ~32 examples at once. Scrolling through this list to find something was easy. It was easy both because I could see more and because it was easy to see collections, for example all loaders started with "loader", all lighting started with "lights" and as a list of 32 text lines it was easy to visually see the grouping. That's lost with thumbnails.


Code_AdtxLxNf4U

Now I can only see 4 at once and there only useful text is the description between each thumbnail. being able to only see 4 at once is vastly less useful but even if they were in a grid it's just much more easy to scan for solutions based on the descriptions than the thumbnails and further it's easier to scan those descriptions as a list of text without the thumbnails.


aQlEKplEm2

**Comment by looeee** [comment]
> I have no idea. The useful info is the text description, not the thumbnail

I guess different people's minds work in different ways. I remember better visually, and best of all when it's text and image combined. 

Please don't revert this. For me it's a huge UI improvement, and I suspect it's an even bigger improvement for beginners, for whom the titles don't hold as much meaning. 

The biggest issue is the size of the list, which makes it hard to quickly scroll to the exact example you want. However, we can continue to work on this. 

Using the search function helps - if you search "loaders" or "geometry", for example, the list becomes manageable.

Maybe we could improve discoverability by adding selectable categories (loader, geometry, webVR etc.)? 

Here's a mockup:

2020-05-28 12_50_35-three js examples

Then when you select Loaders, the search is simply filled with the word "loader".

**Comment by greggman** [comment]
I disagree that it's more useful for beginners. Beginners come looking for solutions. 



Is not a solution. It's just a picture with no meaning.

Thumbnails don't lead to solutions they're just pretty but distracting pictures.

If you want thumbnails I suggest you do something like this

Screen Shot 2020-05-28 at 15 20 16

you can get your pretty pictures but still make the page more useful. When someone clicks a thumbnail you can hide the thumbnails and if you want insert a little "[<]" icon on the bottom left or to go back to the thumbnails or an "[X]" for close

**Comment by Mugen87** [comment]
TBH, I also think that the pure text-based list was more informative and way easier to scan through. As a quick workaround, I vote to introduce a check box to control the display of images (with no images as default).

**Comment by greggman** [comment]
If you want another reason to revert, the current live site is broken on mobile

**Comment by Mugen87** [comment]
> the current live site is broken on mobile

Um, it's seems to look fine on my Pixel. What exactly is broken on your device?

**Comment by greggman** [comment]
If I size the window down on desktop the content disappears

ScreenFlow-size

So checking my iPhone I see the same thing

IMG_3116

**Comment by greggman** [comment]
Looking at the code on the page it's full of hacks, probably piled up over 10+ yrs? Given that it only works in modern browsers since all the samples it points to are using es6 modules, maybe it's time to start over.

Would be happy to try to clean it up using flexbox and other things that all modern browsers support.

**Comment by greggman** [comment]
haha, ok. I guess the blank screen is expected. I would have provided a list to start rather than rely on finding the menu

**Comment by Tirzono** [comment]
I completely agree that the images are not useful for me. I do think the quick fix suggested by @mrdoob to add a checkbox is the best solution for now, but in the long term I would be happy to see the solution suggested by @greggman with the tiled view in the main view and the list view in the menu on the left. UX-wise that feels the most natural.

**Comment by greggman** [comment]
Here's a live version of my hacked up suggested UX

https://raw.githack.com/greggman/three.js/lots-o-thumbs/examples/index.html

ScreenFlow

the change is [here](https://github.com/mrdoob/three.js/compare/dev...greggman:lots-o-thumbs)

**Comment by Mugen87** [comment]
I think the text overlay on top of the image looks a bit cluttered.

image

I'm not a designer so I can't make a suggestion what other solution might look better.

**Comment by greggman** [comment]
Well here's one with the text at the bottom



The issue there is some text is long so it wrap to say 3 or even 4 lines. So then you have to decide what to do

* make all cards have 4 lines (looks like a lot of wasted space)

* make it not a grid (more like pinterest)

  That's not that useful IMO for alphabetized thumbnails because you can't really tell which way things are sorted.

* truncate pass 2 lines

* align with bottom so past 2 lines bleed up into image

* other...

**Comment by Mugen87** [comment]
Well, I've always liked the minimalistic style of the example site 😇 . Playing around with images feels a bit like over-engineering to me. Not really necessary and makes the site only harder to maintain...

**Comment by greggman** [comment]
>  makes the site only harder to maintain...

agreed 

But in any case here's text at the bottom live

https://raw.githack.com/greggman/three.js/fruit-on-the-bottom/examples/index.html

**Comment by marcofugaro** [comment]
Thanks for the mockup @greggman!

That solution was proposed before, however we didn't go with that because the navigation between grid and the example would be awkward, having a list of examples with thumbnails always visible was originally proposed in #18714, and is much more usable and navigable. Also your proposed solution may confuse the user showing two entities of the same example at the same time, leading the user to think they may be different.

-------------------------------

All the other issues you did bring up are all addressed in the proof of concepts I did, please see the description of this PR.

Try to compare the old list view with the all of the changes I proposed in the description of this PR, not the current view. I agree with you that the current implementation is not as navigable as a simple list. 🙂

Also please try to be more open and look at the website with the eyes of a newcomer instead of a person who has used and is used to the list design for ages. I think that will all of the proposed solutions, it becomes more usable than the old list view 🙂


>  the important part is not the image, the important part is what feature/tech/solution is being demonstrated. That info is in the text, not the thumbnail


See the **Tags** and **Sticky section titles** proposals, imagine the category `webgl` would be segmented into detailed categories.

> It was easy both because I could see more and because it was easy to see collections

Can also be solved with more detailed categories and the **Grid view** (#19438)

**Comment by greggman** [comment]
> also please try to be more open 

How about you try to be more open and instead of blindly parroting some unproven idea that images are better maybe open your mind and actually try to reason through what a user on the site is looking for and how they find it. If they want images that lead to random demos the home page does that. Once they get to the example page though they are now looking for solutions, not random thumbnails with no meaning

I'm entirely confident, in fact I'll bet you $5000 in real money if we setup a UX lab and ask 100 devs to look at the page and figure how to do accomplish some task they'd find the answer faster on the text list than the thumbnail list.

To iterate. This thumbnail has **ZERO INFO** that it has anything to do with picking



Similarly this thumbnail has **ZERO INFO** that it has to do with normals



The site isn't a photo site. Photos in a grid help me find a dog or a cat. They don't help me find an example on extrusion or post processing or indexedbuffergeometry because there is no picture that can convey those concepts.

Categories are not a solution. That's just putting lipstick on a pig.

**Comment by Tirzono** [comment]
> Try to compare the old list view with the all of the changes I proposed in the description of this PR, not the current view. I agree with you that the current implementation is not as navigable as a simple list. 🙂

This feels weird to me. Why implementing something that is not as navigable as a simple list? That sounds like a step back. Isn't the purpose of a menu not to make it easy to navigate around?

> Also please try to be more open and look at the website with the eyes of a newcomer instead of a person who has used and is used to the list design for ages. I think that with all of the proposed solutions, it becomes more usable than the old list view 🙂

I do appreciate you are looking into ways to make it easier for newcomers to get around this library. I wasn't aware newcomers had problems with navigating the examples (do they have problems with the list?). But I highly doubt this is the way to go to. I have never seen an example site, or any navigation system, like this. Having to scroll is a bad user experience.

So apparently the reason why we deployed this is because we want to make it easier for newcomers. What about the not-newcomers, people like me that use the library for a long time already? This new change will slow me down a lot. I get it's personal, but for me a list works much better.

**tldr**: I highly doubt the new thumbnail navigation is better than the list navigation. But at least bring back the option to switch it back to the list navigation if a user wants that.

**Comment by WestLangley** [comment]
@greggman How about

https://raw.githack.com/greggman/three.js/fruit-on-the-bottom/examples/index.html

...but without the text under each screenshot?

**Comment by mrdoob** [comment]
@greggman @Tirzono 

Many thanks for the feedback 🙏
The good news is that this is something that can be easily tweaked. 

Would it be possible for you to explain how do you use the list? I personally tend to use the search (regardless of the screenshot) so I rarely scroll.

What was the last item you were looking for when you were scrolling?

**Comment by mrdoob** [comment]
## Postmortem

### Summary

@callumprentice [proposed it](https://github.com/mrdoob/three.js/issues/17692) a few times and he got upset that I didn't give feedback. I didn't think this was something that was needed but I tried to be supportive and [suggested for ways I thought it could work](https://github.com/mrdoob/three.js/issues/17692#issuecomment-544135817).

@looeee seemed interested in the idea, @munrocket saw the screenshots he was doing for e2e could be reused and he also proposed [his own ui](https://github.com/mrdoob/three.js/pull/18839) which its own set of disputes.

Lastly, @marcofugaro came to the rescue and tried to integrate all this with the current design 🙏

I still  think the feature has value, but it may be good to have it behind a checkbox.

### Conclusion

If someone submits a PR with a code change, it's easy to judge, either it works or it doesn't.

The open source model is always a pain when it comes to design/ux work. I already learnt this from when we tried to come up for a logo for the project and everyone started to propose a cube as a logo.

Giving feedback about logo design, ui design, ux design is not that simple and people get upset when they get push backs. They don't want to accept that graphic design requires a lot of practice.

Based on these experiences, I'm tempted to no longer accept PR related to design.

**Comment by callumprentice** [comment]
@mrdoob - with respect, I don't think that accurately describes what happened. 

I proposed an implementation that had a visual preview because I felt that it would be useful. I wrote a version that looks identical to the old one except there was a **checkbox** to show/hide thumbnails next to the search magnifying glass. You didn't like that and suggested I make a version with the thumbnails in the main area like the showcase on the main page.  I made 2 or 3 more example sites based on that and waited for feedback only to be told by you that "there is little chance of that being accepted - why don't you consider something like the current design but with checkboxes" - exactly what I first proposed.

That was a little disheartening after all the work I put in - plus, my scripts to capture screenshots and ensure something visually relevant was captured seems to have been ignored.

I know (and others) you put an unimaginable amount of time into this project which is immensely appreciated. I've had so much fun experimenting with three.js over the years that I wanted to contribute something back and this seemed like the perfect opportunity.

What did I do poorly that I can do differently next time?  Perhaps discuss the design up front in the proposal issue before doing any work? What would make your life easier?

**Comment by mrdoob** [comment]
Sorry for oversimplifying the evolution of this issue.

> What did I do poorly that I can do differently next time? Perhaps discuss the design up front in the proposal issue before doing any work?

Yes, that's probably the solution. Doing a mockup in Photoshop before building it and see what the community thinks.

**Comment by callumprentice** [comment]
Understood @mrdoob and will do.

Since I made the initial proposal, it's my fault this whole divisive issue came about and I apologize. 

For what it's worth, I didn't necessarily think it would be a good idea for beginners - somewhat selfishly, I often found myself remembering an example I'd played with, thinking it would be interesting to inspect the code but not remembering the name and being able to scan them visually might be handy. 

In the future, I'll be ~more conscious of~ never consider trying to bring nonexistent design skills to bear :)

**Comment by WestLangley** [comment]
>I often found myself remembering an example I'd played with, thinking it would be interesting to inspect the code but not remembering the name and being able to scan them visually might be handy.

Agreed! I used http://www.realtimerendering.com/threejs/

I think it would be best to just have the thumbnails (screenshots) on a separate page.

**Comment by greggman** [comment]
> navigation between grid and the example would be awkward

Clicking a thumbnail and having a close button is no different than just about 90% of all photos sites. [Google Photos](https://photos.google.com/album/AF1QipPJNPvHiDZEwVctuhhulBb7u1PZzuLzSBTxCk3y) , [flickr](https://www.flickr.com/photos/greggman/albums/72157712154217288), [Facebook](https://www.facebook.com/pg/upsdogs/photos/?tab=album&album_id=494104860679783), [SmugMug](https://fwphotography.smugmug.com/Pet-Photography/Cats/Cat-Portraits/). Facebook was 2 billion users and somehow navigating between a grid and the actual thing to view seems to be fine. 

>  it may confuse the user showing two entities of the same example at the same time

First off users are not idiots. Also btw youtube has 2 lists



So does iTunes

Screen Shot 2020-05-30 at 14 06 40



I think it's important to remember who is the target audience. This is a site about a **JavaScript Library**. The front page has thumbnails that lead to pretty demos made with the library so if some non-programmer comes by they can click the pretty pictures and see cool stuff, But, the rest of **the site is for JavaScript programmers trying to use a JavaScript library**.

Would MDN better if instead of the list on the left each article had a thumbnail?

 

How about MSDN

 

I'm sure someone here will think that would be an improvement but just remember, the thumbnails will more likely than not suggest nothing about what the actual topic is. Especially if they are auto-generated.

**Comment by Tirzono** [comment]
@mrdoob I couldn't really remember what I was last looking for while scrolling, so I went through the list with thumbnail on the website. It took me a couple of minutes of scrolling when I gave up about remembering what the last thing would be. What maybe would be more interesting is that more often I found some useful feature by scrolling randomly or looking for something particular (GLTF loader as I can remember) and I found the other loaders listed there as well, giving me an idea of what type of other loaders are there.

And speaking of the loader examples: the thumbnails there make no sense at all. It just shows some random object that it loaded with that specific loader, which says nothing about the functionality the example tries to show.

To add to @greggman's post: I completely agree with him and the reason why lists work is because people are used to it. It's the same as making the submit button red: maybe it looks nice, but people are used to have it being green. Or a door where the handle goes up to open the door instead of down: fancy, but not functional. That doesn't mean that I'm against new ideas, but I prefer following the "bigger companies/websites" that pump a lot of money in UI/UX. Those solutions feel natural.

I would be more than happy to make a checkbox/toggle PR. Even when it won't get merged then at least I can host the documentation my own so that people who prefer the old list can work with it again.

**Comment by munrocket** [comment]
Probably worth mentioning a solution from Unity



First mockup by Callumprentice and my first attempt with [searching-ball](https://munrocket.github.io/three.js/examples/) not touched previous list with examples too. Different people have different mindsets and even if somebody think that images not describe anything we have another group of people that remember pictures because they visuals. 

So the best solution that will fit to everybody:

* save list of example names, because it's compact
* save searching by screenshots, because we wanted it too
* improve tag system that visible in UI and easy to maintain (current system more like aliases)

Keeping this diversity in searching process is important when we want to improve system to all situations. Also was idea to create a filter by date/release, to quickly find new examples in fresh release.

**Comment by munrocket** [comment]
By the way, card design from Marcofugaro looks better than fading effect, at least to me.

**Comment by marquizzo** [comment]
I find the thumbnails to be very useful to remember demos that I've seen in the past. That being said, @greggman this is such an elegant solution! Keeping the list as it's always been, while adding a thumbnail grid for those like me who need it. In retrospect, it seemed like the path of least resistance.

Screen Shot 2020-05-28 at 15 20 16

The `/examples/` landing page right now has so much empty real estate, it's just a sea of white that could be put to better use.

**Comment by looeee** [comment]
> a sea of white that could be put to better use.

One suggestion that I liked was to put a rotating three.js logo there. Functionally, though, the grid of examples beside the list on the left is the best choice _on large screens_. How would that look on mobile? The current long list with screenshots works great on small screens.

**Comment by greggman** [comment]
The grid is responsive

IMG_3166

IMG_3167

**Comment by mrdoob** [comment]
Here's one solution to the current design: #19881

image
